### DISCUSSION TITLE
New way to calculate if we passed a section

### DISCUSSION THREAD

**Discussion by MTGPROD** [ppy/osu#13115] [Ideas]
**Describe the new feature:** 

So, it seems to me that the current system only looks at whether our life bar is above 50% or not to tell if we've passed a section. But I'd like to propose something a little more complex to find out if we've passed a section or not. So instead of looking at our life bar, I think we should take the maximum number of MISS or 50 (or even combine both) that we can do in a section and take the number of MISS that the user has done during the section and from that make a percentage. If the percentage is higher than 30-40% we say that the section is partially passed, and above 50-60% we consider the section is failed. (This could also add a skinning element).

**Proposal designs of the feature:**

**Comment by mcendu** [comment]
The rank can be a good measure.

**Comment by mcendu** [comment]
Also it seems that osu!lazer has disposed the concept of section pass/fail.

**Comment by bdach** [comment]
If you're talking about the lack of support for storyboard pass/fail layers, [that is yet to be re-implemented](https://github.com/ppy/osu/issues/6842).

**Comment by MTGPROD** [comment]
@mcendu The rank should be used at the end of the map. For a overall of the performance.

**Comment by MandL27** [comment]
If we're using section grade for section pass/fail, it should be shown on the break screen along with cumulative acc/rank/grade, which is already there. Could also give more granular feedback on song performance, since section grades better convey where players struggle.

Quick and dirty mockup for result screen with section grades:
![image](https://user-images.githubusercontent.com/10366615/85909587-ab275a00-b7e8-11ea-8f74-9c55968518fb.png)

Same, but for break screen:
![image](https://user-images.githubusercontent.com/10366615/85909877-09087180-b7ea-11ea-9f4e-60265934c379.png)

**Comment by peppy** [comment]
Seems like an interesting direction for sure.
